### PR TITLE
Dates must be translated in selected language - Closes #2354

### LINKED ISSUE #2354: Dates must be translated in the selected language
As a user; I must be able to see the dates in my language of choice.

### Acceptance criteria
Month names must match the active language in:
 - Transaction details page.
 - Home page, transactions list.
 - Other wallets, transactions list.
 - Community feed
 - Home, Balance chart
 - Other wallet, balance chart
 - Delegate profile

### PR DESCRIPTION
**Pull request by massao**
### What issue have I solved?

#2354 

#### Using Moment
- [x] Transaction details page.
- [x] Wallet page, transactions list.
- [x] Other wallets, transactions list.
- [x] Wallet, Balance chart
- [x] Other wallet, balance chart
- [x] Delegate profile
- [x] Calendar datepicker on transactions filter

#### Needs translation on common.json
- [x] Dashboard -> Community feed
- [x] Transactions list -> to show [today and yesterday] correctly.

I updates strings that were missing the `t()` function, and updated places using moment.js to consider the selected language  
Moment.js already has localization, only issue is with the calendar function, used to generate string as `[Today] DD.MM.YYYY`, that should be translated, and also the date formats updated if needed.

### How has this been tested?

It's needed to set a `feature-flag-language` into localStorage to see the option in settings. 
```js
localStorage.setItem('feature-flag-language', true);
```

1. Go to Settings;
2. Change language;
3. See that dates are shown with German month names.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/transactions/filters/filterBar.js [hunk 1]
```diff
@@ -1,57 +1,62 @@
 import React from 'react';
 import moment from 'moment';
+import 'moment/min/locales';
```
Comment (slaweet) [inline src/components/transactions/filters/filterBar.js]:
```js
import moment from 'moment/min/moment-with-locales';
```
looks cleaner
https://stackoverflow.com/a/52559157/2080716

File: i18n/locales/de/common.json [hunk 2]
```diff
@@ -229,5 +229,10 @@
   "{{count}} of the delegate names selected for voting was already voted for for:": "{{count}} der ausgewählten Delegierten-Namen zum Wählen sind bereits gewählt:",
   "{{count}} of the delegate names selected for voting was already voted for for:_plural": "{{count}} der ausgewählten Delegierten-Namen zum Wählen sind bereits gewählt:",
   "{{count}} of the provided delegate names could not be resolved:": "{{count}} der eingegebenen Delegierten-Namen konnten nicht gefunden werden:",
-  "{{count}} of the provided delegate names could not be resolved:_plural": "{{count}} der eingegebenen Delegierten-Namen konnten nicht gefunden werden:"
+  "{{count}} of the provided delegate names could not be resolved:_plural": "{{count}} der eingegebenen Delegierten-Namen konnten nicht gefunden werden:",
+  "[Today], hh:mm A": "[Heute], hh:mm A",
+  "[Tomorrow], hh:mm A": "[Morgen], hh:mm A",
+  "[Yesterday], hh:mm A": "[Gestern], hh:mm A",
+  "{{days}}d ago": "Vor {{days}}d",
+  "{{hours}}h ago": "Vor {{hours}}h"
```
Comment (slaweet) [inline i18n/locales/de/common.json]: These strings are missing from `i18n/locales/en/common.json`. Looks like they are not automatically generated, which will be a problem.

File: src/components/transactions/filters/dateFieldGroup.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable max-lines */
```
Comment (slaweet) [inline src/components/transactions/filters/dateFieldGroup.js]: 😢 
Maybe you can save some lines by replacing `this.handleClickOutsideDropdown` with use of `DropdownButton` 😉 
Or open a refactoring ticket for this file.

Comment (massao) [reply src/components/transactions/filters/dateFieldGroup.js]: I think that the creating a refactor ticket is better, so this PR doesn't go to far from the original scope.

File: i18n/locales/de/common.json (hunk 2 above)
Comment (massao) [reply i18n/locales/de/common.json]: Oh actually these were already on the `en/common.json`, I just added  to `de/common.json`,  to be able to test.
https://github.com/LiskHQ/lisk-hub/blob/6d4512c60680992a49c079a5ded1425855856396/i18n/locales/en/common.json#L521-L531

Not exactly same order, as I just added them to the end of file.

Comment (slaweet) [reply i18n/locales/de/common.json]: Ah, got it now 👍

File: src/components/transactions/filters/dateFieldGroup.js (hunk 3 above)
Comment (massao) [reply src/components/transactions/filters/dateFieldGroup.js]: issue created: #2360

**Pull request merged**